PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,650 @@
+    private let callbackQueue = DispatchQueue(label: "Backend callbackQueue")
+    private var offeringsCallbacksCache: [String: [OfferingsResponseHandler]]
+    private var purchaserInfoCallbacksCache: [String: [BackendPurchaserInfoResponseHandler]]
+
+    private var authHeaders: [String: String] { return ["Authorization": "Bearer \(self.apiKey)"] }
+
+    @objc(initWithAPIKey:systemInfo:eTagManager:operationDispatcher:)
+    public convenience init(apiKey: String,
+                            systemInfo: SystemInfo,
+                            eTagManager: ETagManager,
+                            operationDispatcher: OperationDispatcher) {
+        let httpClient = HTTPClient(systemInfo: systemInfo,
+                                    eTagManager: eTagManager,
+                                    operationDispatcher: operationDispatcher)
+        self.init(httpClient: httpClient, apiKey: apiKey)
+    }
+
+    @objc public required init(httpClient: HTTPClient, apiKey: String) {
+        self.httpClient = httpClient
+        self.apiKey = apiKey
+        self.offeringsCallbacksCache = [:]
+        self.purchaserInfoCallbacksCache = [:]
+    }
+
+    @objc(createAliasForAppUserID:newAppUserID:completion:)
+    public func createAlias(appUserID: String, newAppUserID: String, completion: ((Error?) -> Void)?) {
+        guard let appUserID = try? escapedAppUserID(appUserID: appUserID) else {
+            completion?(ErrorUtils.missingAppUserIDError())
+            return
+        }
+
+        let path = "/subscribers/\(appUserID)/alias"
+        httpClient.performPOSTRequest(serially: true,
+                                      path: path,
+                                      requestBody: ["new_app_user_id": newAppUserID],
+                                      headers: authHeaders) { statusCode, response, error in
+            self.handle(response: response, statusCode: statusCode, error: error, completion: completion)
+        }
+    }
+
+    @objc public func clearCaches() {
+        httpClient.clearCaches()
+    }
+
+    // swiftlint:disable function_parameter_count
+    @objc (postReceiptData:appUserID:isRestore:productInfo:presentedOfferingIdentifier:observerMode:subscriberAttributes:completion:)
+    public func post(receiptData: Data,
+                     appUserID: String,
+                     isRestore: Bool,
+                     productInfo: ProductInfo?,
+                     presentedOfferingIdentifier offeringIdentifier: String?,
+                     observerMode: Bool,
+                     subscriberAttributes subscriberAttributesByKey: SubscriberAttributeDict?,
+                     completion: @escaping BackendPurchaserInfoResponseHandler) {
+    // swiftlint:enable function_parameter_count
+        let fetchToken = receiptData.base64EncodedString(options: .lineLength64Characters)
+        var body: [String: AnyObject] = [
```

Comment:
I think the conversion into `NSObject`s isn't needed now that `HTTPClient` is implemented in Swift

Reply:
we can also go ahead and make the `HTTPClient` `internal` and remove the `@objc` modifiers from its methods

Reply:
Will do it in another PR